src/terraingenerator/main.cpp

```diff
@@ -319,27 +460,31 @@ int main(int argc, char *argv[])
             if (!terrains.contains(terrain->name()))
                 terrains.insert(terrain->name(), terrain);
 
-    // Assign terrain priority based on the order in which we encounter
-    // them (this will determine drawing order).
-    QStringList terrainOrder;
-    terrainOrder << "LavaRock";
-    terrainOrder << "Dirt2";
-    terrainOrder << "Dirt";
-    terrainOrder << "Sand";
-    terrainOrder << "Water";
-    terrainOrder << "Hole";
-    terrainOrder << "Grass";
-    terrainOrder << "PlowedSoil";
-    terrainOrder << "Lava";
+    // Setup terrain priorities.
     TerrainLessThan lessThan;
     int priority = 0;
-    foreach (const QString &terrainName, terrainOrder) {
+    foreach (const QString &terrainName, options.terrainPriority) {
         lessThan.terrainPriority.insert(terrainName, priority);
-        ++priority;
+        priority++;
     }
 
     qDebug() << "Terrains found:" << terrains.keys();
 
+    // Check if all terrains from priority list were found and loaded.
+    foreach (const QString &terrainName, lessThan.terrainPriority.keys())
+        if (!terrains.contains(terrainName))
+            qWarning() << "Terrain" << terrainName << "from priority list not found.";
```

Comment:
This is a nice check, but it is actually more important to make sure that all terrains mentioned in combine lists are present. It looks like the tool would currently crash in such a case.